plugins/moduleLoader.js

```diff
@@ -2,89 +2,125 @@
+import additionalConfig from '@Root/modules.config'; // User modules configuration file
 
-const modules = [];
+class ModuleLoader {
+    constructor() {
+        this.config = this.setModulesConfiguration();
+        this.pagesConfig = this.setPagesConfiguration(this.config);
+        this.componentsConfig = this.setComponentsConfiguration(this.config);
+    }
+
+    get getPagesConfig() {
+        return this.pagesConfig;
+    }
+
+    get getComponentsConfig() {
+        return this.componentsConfig;
+    }
 
-const ModuleLoader = {
     install() {
-        this.getData().then((components) => {
-            components.forEach((component) => {
-                this.getComponent(component.path, component.module);
-            });
-        });
-    },
-    getRouterConfigs() {
-        return modules.reduce((previousObject, currentObject) => {
+        const { router, menu } = this.pagesConfig;
+        Vue.prototype.$modulesConfiguration = {
+            router,
+            menu,
+        };
+    }
+
+    requireForce(modulePath) { // force require
+        try {
+            return require(`${modulePath}`);
+        } catch (e) {
+            return false;
+        }
+    }
+
+    setModulesConfiguration() {
+        let summaryConfig = { pages: [], components: [] };
+        const {
+            pages: additionalConfigPages,
+            components: additionalConfigComponents,
+        } = additionalConfig;
+
+        const modulesConfig = this.requireForce('@Root/config/modules'); // File may not exist, build from CLI
+        if (modulesConfig) {
+            const { pages: modulePages, components: moduleComponents } = modulesConfig.default;
+            summaryConfig = {
+                pages: [...modulePages],
+                components: [...moduleComponents],
+            };
+        }
+        summaryConfig.pages.push(...additionalConfigPages);
+        summaryConfig.components.push(...additionalConfigComponents);
+        return summaryConfig;
+    }
+
+    setPagesConfiguration({ pages }) {
+        const pagesConfiguration = { router: [], menu: [], store: [] };
+
+        for (let i = 0; i < pages.length; i += 1) {
             let config = null;
-            switch (currentObject.moduleSource) {
-            // TODO: commented because when there is no directory for an alias, nuxt returns an error.
-            // TODO: uncomment if  we have modules to load
-            // case 'vendor':
-            //     config = require(`@Vendors/${currentObject.name}/config`); // eslint-disable-line global-require, import/no-dynamic-require
-            //     break;
-            // case 'module':
-            //     config = require(`@Modules/${currentObject.name}/config`); // eslint-disable-line global-require, import/no-dynamic-require
+            const { isActive, name, source } = pages[i];
+            const pageName = `pages/${name}`;
+
+            if (!isActive) continue; // eslint-disable-line no-continue
```

Comment:
I'm not sure about using 'for' with continue with disabled eslint. I think it will be better to use filter (to filter out inactive modules) and then forEach with below logic.

Reply:
I would rather revert for loop logic to avoid using continue